src/v/config/configuration.cc

```diff
@@ -1483,6 +1483,14 @@ configuration::configuration()
       "threshold the cloud storage is considered being idle.",
       {.needs_restart = needs_restart::no, .visibility = visibility::tunable},
       1.0)
+  , cloud_storage_background_jobs_quota(
+      *this,
+      "cloud_storage_background_jobs_quota",
+      "The number of total requests that the cloud storage background jobs are "
+      "allowed to make during one background housekeeping run. This is a per "
+      "shard limit.",
+      {.needs_restart = needs_restart::no, .visibility = visibility::tunable},
+      5000)
```

Comment:
Given we expect at least some of these tasks to be tied to partitions, I wonder if we should be accounting for the total number of partitions

Reply:
I don't think it's necessary. We just want to ensure that the background jobs make some progress. If there's loads of queued jobs due to high partition counts, we'll process them in the next iteration. If we added a per partition quota, we would still need some sort of aggregated max.